service/wave/wave.go

```diff
@@ -0,0 +1,334 @@
+	SparkApplications []*SparkApplication `json:"sparkApplications,omitempty"`
+}
+
+type ReadSparkApplicationInput struct {
+	ID *string `json:"id,omitempty"`
+}
+
+type ReadSparkApplicationOutput struct {
+	SparkApplication *SparkApplication `json:"sparkApplication,omitempty"`
+}
+
+func (s *ServiceOp) ListClusters(ctx context.Context, input *ListClustersInput) (*ListClustersOutput, error) {
+	r := client.NewRequest(http.MethodGet, "/wave/cluster")
+
+	if input != nil {
+		if input.ClusterIdentifier != nil {
+			r.Params.Set("clusterIdentifier", spotinst.StringValue(input.ClusterIdentifier))
+		}
+
+		if input.ClusterState != nil {
+			r.Params.Set("state", spotinst.StringValue(input.ClusterState))
+		}
+	}
+
+	resp, err := client.RequireOK(s.Client.Do(ctx, r))
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	clusters, err := clustersFromHttpResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ListClustersOutput{Clusters: clusters}, nil
+}
+
+func (s *ServiceOp) ReadCluster(ctx context.Context, input *ReadClusterInput) (*ReadClusterOutput, error) {
+	if input == nil {
+		return nil, fmt.Errorf("input is nil")
+	}
+
+	path, err := uritemplates.Expand("/wave/cluster/{clusterId}", uritemplates.Values{
+		"clusterId": spotinst.StringValue(input.ClusterID),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	r := client.NewRequest(http.MethodGet, path)
+	resp, err := client.RequireOK(s.Client.Do(ctx, r))
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	clusters, err := clustersFromHttpResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	output := new(ReadClusterOutput)
+	if len(clusters) > 0 {
+		output.Cluster = clusters[0]
+	}
+
+	return output, nil
+}
+
+func (s *ServiceOp) DeleteCluster(ctx context.Context, input *DeleteClusterInput) (*DeleteClusterOutput, error) {
+	if input == nil {
+		return nil, fmt.Errorf("input is nil")
+	}
+
+	path, err := uritemplates.Expand("/wave/cluster/{clusterId}", uritemplates.Values{
+		"clusterId": spotinst.StringValue(input.ClusterID),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	r := client.NewRequest(http.MethodDelete, path)
+
+	if input.ShouldDeleteOcean != nil {
+		r.Params.Set("shouldDeleteOcean",
+			strconv.FormatBool(spotinst.BoolValue(input.ShouldDeleteOcean)))
+	}
+
+	resp, err := client.RequireOK(s.Client.Do(ctx, r))
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	return &DeleteClusterOutput{}, nil
+}
+
+func (s *ServiceOp) ListSparkApplications(ctx context.Context, input *ListSparkApplicationsInput) (*ListSparkApplicationsOutput, error) {
+	r := client.NewRequest(http.MethodGet, "/wave/spark/application")
+
+	if input != nil {
+		if input.ClusterIdentifier != nil {
+			r.Params.Set("clusterIdentifier", spotinst.StringValue(input.ClusterIdentifier))
+		}
+
+		if input.Name != nil {
+			r.Params.Set("name", spotinst.StringValue(input.Name))
+		}
+
+		if input.ApplicationId != nil {
```

Comment:
```suggestion
		if input.ApplicationID != nil {
```
[Ref.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)